File: app/src/main/java/com/example/kyuyeol/onedaydrink/MainActivity.java
package com.example.kyuyeol.onedaydrink;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.support.constraint.ConstraintSet;
import android.support.transition.TransitionManager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

import com.skt.Tmap.TMapView;

public class MainActivity extends AppCompatActivity {

    FrameLayout view;
    ConstraintLayout constraintLayout;
    View bottomBar;
    ConstraintSet constraintSet1;
    ConstraintSet constraintSet2;

    boolean bottomState;

    RecyclerView recyclerView1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        view = (FrameLayout)findViewById(R.id.Tmap);
        constraintLayout = (ConstraintLayout)findViewById(R.id.constraint);
        bottomBar = (View)findViewById(R.id.bottom_bar);
        recyclerView1 = (RecyclerView)findViewById(R.id.recyclerview);

        TMapView tmapview = new TMapView(this);
        tmapview.setSKTMapApiKey("df15431c-c688-49f4-b53a-6e5f56f0ed90");
        view.addView(tmapview);

        constraintSet1 = new ConstraintSet();
        constraintSet1.clone(constraintLayout);
        constraintSet2 = new ConstraintSet();
        constraintSet2.clone(this, R.layout.activity_main2);

        bottomState = false;
        bottomBar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d("TAG", "TAGGGGGG");
                TransitionManager.beginDelayedTransition(constraintLayout);
                if(bottomState) constraintSet1.applyTo(constraintLayout); else constraintSet2.applyTo(constraintLayout);
                bottomState = !bottomState;
            }
        });

        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        recyclerView1.setLayoutManager(layoutManager);
        recyclerView1.setAdapter(new RecyclerViewAdapter(this));

    }

    @Override
    public void onBackPressed() {
        if(!bottomState) super.onBackPressed();
        else {
            TransitionManager.beginDelayedTransition(constraintLayout);
            constraintSet1.applyTo(constraintLayout);
            bottomState = !bottomState;
        }
    }
}

File: app/src/main/java/com/example/kyuyeol/onedaydrink/MainActivity.kt
package com.example.kyuyeol.onedaydrink

import android.graphics.Color
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.view.LayoutInflater
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.fragment_tabview.view.*
import java.util.zip.Inflater

class MainActivity : AppCompatActivity(), TabLayout.OnTabSelectedListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tablayout.addTab(tablayout.newTab().setCustomView(R.layout.fragment_tabview))
        tablayout.addTab(tablayout.newTab().setCustomView(R.layout.fragment_tabview))
        tablayout.addTab(tablayout.newTab().setCustomView(R.layout.fragment_tabview))
        tablayout.addTab(tablayout.newTab().setCustomView(R.layout.fragment_tabview))

        CustomTab(0)
        CustomTab(1)
        CustomTab(2)
        CustomTab(3)

        val adapter = TabPagerAdapter(supportFragmentManager, tablayout.tabCount)

        viewpager.adapter = adapter

        viewpager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tablayout))
        tablayout.addOnTabSelectedListener(this)

    }

    override fun onTabReselected(tab: TabLayout.Tab?) {

    }

    override fun onTabUnselected(tab: TabLayout.Tab?) {

    }

    override fun onTabSelected(tab: TabLayout.Tab) {
        viewpager.currentItem = tab.position
    }

    fun CustomTab(position: Int) {
        val view: View? = tablayout.getTabAt(position)?.customView
        when (position) {
            0 ->view?.tab_text?.text = "Test1"
            1 -> view?.tab_text?.text = "Test2"
            2 -> view?.tab_text?.text = "Test3"
            3 -> view?.tab_text?.text = "Test4"
        }
    }

}
